alacritty_terminal/src/term/mod.rs

```diff
@@ -1457,6 +1457,57 @@ impl<T> Term<T> {
             rendered: false,
         }
     }
+
+    fn grid_to_string_between(&self, y0: Line, y1: Line) -> String {
+        let grid = &self.grid;
+
+        let mut s = String::with_capacity(*(y1 - y0) * *grid.cols());
+
+        for y in (*y0..*y1).rev() {
+            let row = &grid[y];
+            let wrapline = row.last().unwrap().flags.contains(Flags::WRAPLINE);
+            let mut line_len = 0;
+
+            // Calculate the line length. We need to do this to avoid trailing spaces.
+            if wrapline {
+                line_len = *grid.cols();
+            } else {
+                for n in (0..*grid.cols()).rev() {
+                    if row[Column(n)].c != ' ' {
+                        line_len = n + 1;
+                        break;
+                    }
+                }
```

Comment:
This is not the correct way to calculate line length/emptiness.

Reply:
Oh. I found the `LineLength` trait that `grid::Row<Cell>` implements. I am assuming that is the correct way? Please tell me if that is not the case.

Reply:
It is not, sorry for not being more specific here. But checking for cells matching `' '` will cut off whitespace at the end. You should iterote over the cells in reverse and check `!cell.is_empty()`.

Reply:
I think it is correct now.